.pre-commit-config.yaml

```diff
@@ -1,12 +1,22 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
 repos:
--   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v1.4.0
-    hooks:
-    - id: flake8
-      additional_dependencies: [flake8-docstrings, flake8-quotes, flake8-import-order, flake8-annotations]
-    - id: check-added-large-files
-    - id: detect-private-key
-    - id: trailing-whitespace
-    - id: end-of-file-fixer
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v1.4.0
+  hooks:
+  - id: check-added-large-files
+  - id: detect-private-key
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.0.280
+  hooks:
+    - id: ruff
+      args: [ --fix, --exit-non-zero-on-fix ]
+- repo: https://github.com/psf/black
+  rev: 23.7.0
```

Comment:
not sure if this will ever be a problem, but we should file away in our brains the possibility that a pinned version of a pre-commit linter may be older than what we have in CI (looks like the actions use the newest version by default) and what we have installed in our local pipenvs/venvs (whatever the newest version was the last time we did a `pip install`). Like if something may pass local pre-commit but fail in Actions, we'll just need to remember to check that it's not a version discrepancy.